README.md

```diff
@@ -8,9 +8,221 @@
+def myProgram(using context: ExchangeCalculus & ExchangeLanguage): Int =
+  // neighborValues: share a value with neighbors and get their values back
+  val neighborIds = neighborValues(localId)
+  
+  // evolve: maintain state across rounds (like a distributed variable)
+  val roundCount = evolve(0)(count => count + 1)
+  
+  // share: compute values while sharing them with neighbors
+  val consensus = share(0)(neighborValues => 
+    (neighborValues.toIterable.sum + roundCount) / (neighborValues.size + 1)
+  )
+  
+  consensus
+```
+
+### Distance Gradient
+
+A classic aggregate programming example—computing distance from a source:
+
+```scala
+import it.unibo.scafi.libraries.All.*
+
+def gradient(using context: ExchangeCalculus & ExchangeLanguage { type DeviceId = Int }): Double =
+  // Compute distance from source using neighbor distance + edge cost
+  distanceTo[Double, Double](
+    source = localId == 0, // Device 0 is the source
+    distances = neighborValues(1.0) // Each hop costs 1.0
+  )
+```
+
+### Domain Splitting with Branching
+
+Control information flow with domain branching:
+
+```scala
+import it.unibo.scafi.libraries.All.*
+
+def conditionalBehavior(using context: ExchangeCalculus & ExchangeLanguage { type DeviceId = Int }): String =
+  val distanceFromSource = gradient
+  
+  // Split the network into two independent computational domains
+  branch(distanceFromSource < 5.0)(
+    "Close to source: " + distanceFromSource
+  )(
+    "Far from source: " + distanceFromSource
+  )
+```
+
+### Exchange Calculus: The Foundation
+
+Under the hood, ScaFi implements the **Exchange Calculus**, a more expressive variant of field calculus:
+
+```scala
+import it.unibo.scafi.libraries.All.*
+
+def exchangeExample(using context: ExchangeCalculus & ExchangeLanguage): Int =
+  // exchange: send and receive different values
+  exchange(0) { receivedValues =>
+    val maxFromNeighbors = receivedValues.withoutSelf.max
+    val myValue = 1 + maxFromNeighbors
+    
+    // Return one value but send another to neighbors
+    returning(myValue) send (myValue + 1)
+  }(localId)
+```
+
+### Complete Example: Self-Healing Gradient
+
+Here's a complete example showing ScaFi's resilience:
+
+```scala
+import it.unibo.scafi.language.xc.{ExchangeLanguage, ExchangeCalculus}
+import it.unibo.scafi.libraries.All.*
+
+// Define an aggregate program that computes a self-healing gradient
+def selfHealingGradient(
+  isSource: Boolean, 
+  hopDistance: Double = 1.0
+)(using context: ExchangeCalculus & ExchangeLanguage): Double =
+  
+  // Automatically computes and maintains shortest distance from source
+  // Adapts to topology changes and device failures
+  distanceTo[Double, Double](isSource, neighborValues(hopDistance))
+
+// In practice, this would be executed by the ScaFi engine across devices
+// Each device runs the same program but operates on its local context
+```
+
+## Key Concepts
+
+### Shared Data Types
```

Comment:
If I remember correctly, this may also be a single value since it can be used like normal scafi. Am i right @nicolasfara ?